File: core/src/ru/myitschool/dcrawler/entities/skills/patterns/TargetPattern.java
package ru.myitschool.dcrawler.entities.skills.patterns;

import ru.myitschool.dcrawler.entities.skills.Skill;
import ru.myitschool.dcrawler.entities.skills.Target;

public abstract class TargetPattern {

    private Skill skill;

    public TargetPattern(Skill skill) {
        this.skill = skill;
    }

    public abstract Target createTarget(Target target);

    public Skill getSkill() {
        return skill;
    }
}

File: core/src/com/wolg_vlad/dcrawler/entities/skills/patterns/TargetPattern.kt
package com.wolg_vlad.dcrawler.entities.skills.patterns

import com.wolg_vlad.dcrawler.entities.skills.Skill
import com.wolg_vlad.dcrawler.entities.skills.Target

abstract class TargetPattern(val skill: Skill) {
    abstract fun createTarget(target: Target): Target

    abstract fun isValid(target: Target): Boolean
}